### DISCUSSION TITLE
New package metric: Number of contributors

### DISCUSSION THREAD

**Discussion by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#1692] [Ideas]
Related to the work I’m doing on scoring, it would be great if we could get a number of contributors from GitHub’s API.

My thinking is that packages that have more contributors will generally be of higher quality.

**Comment by MaximBazarov** [comment]
I don't think it's a fair metric, people with more friends or more public, or more extravert, aren't necessarily produce better solutions.

**Reply by Sherlouk** [reply]
Counterpoint: the more (active) collaborators a project has, the more likely it's to remain active. A project maintained by one person has a high risk of being abandoned.

It's not about friends, if anything that's why stars is a poor metric as you're just asking who has the biggest follower base. It's a lot more effort to convert follower to contributor!

**Reply by finestructure** [reply]
:+1:

As an example, the [Swift Server Work Group Incubation Process](https://github.com/swift-server/sswg/blob/master/process/incubation.md) requires a minimum of two contributors for incubation.

**Reply by daveverwer** [reply]
Very shortly, thanks to #2060, we will have some stats on this so we can start to see many packages have how many authors across the whole dataset.

I still think it's a good metric for all the reasons everyone has mentioned above

**Reply by MaximBazarov** [reply]
I agree with the point:
> he more (active) collaborators a project has, the more likely it's to remain active. A project maintained by one person has a high risk of being abandoned.

I don't think it's a counterpoint, this metric might be as a metric, so people who want it has access to it, I think it should not liverage the score in any way, since it's very prone to false positives.

**Reply by MaximBazarov** [reply]
To look at it from another angle, say I developed a best solution in the world for the problem at hand, package has some flaws but it's the best so far. It gets 5000 stars and some docs and tests, it has a lot of issues but, I resolve them from time to time. Now with the current system, it's the top 1 package, add to this a couple of copies of the same solution, and we eventually have top 7 packages that always will get 99% of attention.


Now a couple of years passed, there's a new solution, that doesn't have flaws, and so simple so it requires maintenance only major swift releases, there's no way for this package to be even discovered if we don't take the package quality into account.

**Reply by MaximBazarov** [reply]
in short, once you successful, you are the best solution, I don't think it's good for the evolution of the ecosystem.

**Reply by MaximBazarov** [reply]
I'm totally ok if you want to go with the current approach to see if I'm anywhere close to the reality btw.

**Reply by MaximBazarov** [reply]
> 👍
> 
> As an example, the [Swift Server Work Group Incubation Process](https://github.com/swift-server/sswg/blob/master/process/incubation.md) requires a minimum of two contributors for incubation.

I also like that, I think it makes sense to eliminate the "BUS factor"

**Reply by finestructure** [reply]
What I was getting at when saying in Discord that this doesn't compare to Google: you're not picking dependencies every day. Or if you are, you're doing something wrong. And when you do and you stop after looking at the first five, you're also doing it wrong. This is not a job an algorithm can do for you. You have to inspect those results in more depth and in more breadth than you would a shopping query.

I don't believe we should be spending too much time trying to come up with a Google-like page rank (which not everyone will agree with anyway) for those reasons.

Of course we'll try and add more metrics like tests and contributors and augment the score over time. But the point of that is mainly to make the detailed inspection of packages easier rather than come up with a "perfect" page rank.
